### PR TITLE
Use fixed random-seed in the "Build & Test" CI

### PR DESCRIPTION
**Pull request by tobiasdiez**
### :books: Description

Related to #35522 and #35544. Should at least fix the changed codecov between runs.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by mkoeppe** [comment]
I think this is a good solution. The goal of #29935 is still sufficiently served by users' local tests and but other workflow runs (such as the ci-linux runs on release tags). @kliem any objections?

**Comment by tobiasdiez** [comment]
There were also two runs for the push on the dev branch (one for the push, and one for the tag), so I guess we will only see after merging this if it really works then.

**Comment by kliem** [comment]
> I think this is a good solution. The goal of #29935 is still sufficiently served by users' local tests and but other workflow runs (such as the ci-linux runs on release tags). @kliem any objections?

I think the suggested changes are an improvement.

Overall, I still think that if people write a test that claims that something works for any random input of a type, it should. E.g. matrix addition should be entry-wise addition.

However, when I pushed for enforcing the random seed for CI, we still used patchbots that would occasionally fail anyway. Now, we use a more stable CI, I believe. If we are able to eliminate almost all noise from the CI, we should I think. And it's also good, if you don't get affected by problems not related to your PR.

**Review by mkoeppe — APPROVED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -3,6 +3,9 @@ name: Build & Test
 on:
   pull_request:
   push:
+    branches: ['**']
```
Comment (mkoeppe) [inline .github/workflows/build.yml:6]: You could just replace this by 
```
push: 
  branches:
```
(it won't run on tags then)

**Pull request merged**